### PR TITLE
Mhp 2107 organization cleanup

### PR DESCRIPTION
**Pull request by jesusfreak3**

### CODE REVIEW

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <div class="flex justify-between items-stretch">
+    <div class="normal pivot-gray-darker mt3-5 pa4 f5 br2 bg-pivot-gray-5 w-32">
+      <h4 class="mt0 mb4 fw6 tracked ttu">{{ 'ministries.cleanup.archive_alumni' | t }}</h4>
+      <p class="lh-copy w-80 mt0 mb0">{{ 'ministries.cleanup.archive_alumni_description' | t }} </p>
+      <a href class="f6 link pa2 ba b--pivot-dark-blue ihshort tc mt3 dib pivot-dark-blue bold br2 ttu tracked bg-transparent w-100 dim" ui-sref="app.ministries.ministry.people({ orgId: $ctrl.orgId })">{{ 'ministries.cleanup.open_contacts' | t }}</a>
+    </div>
+    <div class="normal pivot-gray-darker mt3-5 pa4 f5 br2 bg-pivot-gray-5 w-32">
+      <h4 class="mt0 mb4 fw6 tracked ttu flex items-center">
+        {{ 'ministries.cleanup.archive_by_date' | t }}
+        <icon-button class='ml0' icon='help'></icon-button>
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: I believe these were intended to be tooltips. We should figure out what content should go in them or remove them. Can you ask Alex about them? Thanks :)

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            class="pa2 input-reset bg-white bw0 pivot-gray-darker w-80 ihshort br2 datepicker"
+            type="date"
+            ng-model="$ctrl.archiveBeforeDate">
+        <button class="f6 pa2 link ba b--pivot-dark-blue ihshort mt3 dib pivot-dark-blue bold br2 ttu tracked bg-transparent w-100 dim" ng-click="$ctrl.archiveBefore($ctrl.archiveBeforeDate)">{{ 'ministries.cleanup.archive_by_date' | t }}</button>
+      </fieldset>
+    </div>
+    <div class="normal pivot-gray-darker mt3-5 pa4 f5 br2 bg-pivot-gray-5 w-32">
+      <h4 class="mt0 mb4 fw6 tracked ttu flex items-center">
+        {{ 'ministries.cleanup.archive_by_inactivty' | t }}
+        <icon-button class='ml0' icon='help'></icon-button>
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: Same tooltip comment.

File: app/assets/javascripts/angular/missionhubApp.routes.js [hunk 3]
```diff
@@ -350,6 +350,15 @@ angular
                 url: '/management',
                 component: 'orgManagement',
             })
+            .state({
+                name: 'app.ministries.ministry.cleanup',
+                url: '/organization-cleanup',
+                component: 'organizationCleanup',
+                resolve: {
+                    orgId: ($state, $transition$) =>
```
[the remark this answers is not in the document]
Comment (jesusfreak3) [reply app/assets/javascripts/angular/missionhubApp.routes.js]: my mistake, copied it from the reportMovementIndicators which looks like it doesnt need it either?  removing that one too.

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    template: template,
+    controller: organizationCleanupController,
+});
+
+function organizationCleanupController(httpProxy, $scope) {
+    this.archiveBeforeDate = new Date();
+    this.archiveInactivityDate = new Date();
+    this.dateOptions = {
+        formatYear: 'yy',
+        maxDate: new Date(2020, 5, 22),
```
[the remark this answers is not in the document]
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js]: yah left over from my other attemps with other date pickers... need to really double check things like this sorry.

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        if (!dateBy) return;
+
+        const { data } = await archiveContacts(
+            'persons_inactive_since',
+            dateBy,
+        );
+
+        if (data) this.cleanupCompleted = true;
+
+        $scope.$apply();
```
[the remark this answers is not in the document]
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js]: yah ok, its cleaner.. I saw the duplicate code, should have refactored earlier

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  <h1 class="f2 pivot-dark-blue fw6 mv4">
+    {{ 'ministries.cleanup.title' | t }}
+  </h1>
+  <p class="pivot-gray-darker f4-5  mb5" bind-dynamic="'ministries.cleanup.description' | t "></p>
+
+
+  <div class='tc ' ng-if="$ctrl.cleanupCompleted">
+    <header class="center tc pt4">
+        <img class="center" ng-src="{{$ctrl.checkIcon}}" style="width:150px; height:150px;">
+        <h1 class="f2 normal pivot-dark-blue mb3 mt3">{{ 'contact_import:step_4.title' | t }}</h1>
```
[the remark this answers is not in the document]
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: yah, i had a bad dream about this one last night lol... how i left that contact import name space in the locale.... alex asked me to copy the deisn from there, so i copied the html, and forgot to replace that one langauge..... my brain knew i did it, so it let me know, just didnt have enough time to do it before your review!

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html (hunk 1 above)
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: your right totally missed the tool tip or link part..

Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: ok alex suggested to hide them for now as we do not have the proper copy

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html (hunk 2 above)
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: ok alex suggested to hide them for now as we do not have the proper copy

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.html]: haha

**Comment by jesusfreak3** [comment]
@OzzieOrca  checking this, i never had multiple orgs before, so never saw that screen! But since you and alex added the Leader Impact i can test this now, so working on it.

**Comment by jesusfreak3** [comment]
@OzzieOrca ok pushed the fix.

File: app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js [hunk 7]
```diff
@@ -42,7 +45,29 @@ function organizationCleanupController(httpProxy, $scope) {
...
+                    icon: () => this.checkIcon,
+                    closeLabel: () =>
+                        $filter('t')('ministries.cleanup.success_cta'),
+                },
+                windowClass: 'pivot_theme',
+                backdrop: 'static',
+                keyboard: false,
+            });
+
+            const result = await modalInstance.result;
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js]: Doesn't look like you're using `result` anywhere.

**Pull request merged**